pkg/app/src/components/TabPanels/Organization/tmpOverview.tsx

```diff
@@ -43,19 +48,36 @@ export function TmpOverview() {
 			if (state) {
 				// The transaction was successful, clear state
 				tmpOrgState?.clearAll()
+				result.events.forEach(({ event: { data, method, section } }) => {
+					if (section === 'control' && method === 'OrgCreated') {
+						setOrgId(data[1].toHex())
+						if (organizationByIdData && !loading) {
+							push(`/organisations/${organizationByIdData?.organization?.[0]?.id}/dashboard`)
+						}
+					}
+				})
 			}
 		},
 		[tmpOrgState.clearAll],
 	)
 
+	useEffect(() => {
+		if (organizationByIdData && !loading) {
+			push(`/organisations/${organizationByIdData?.organization?.[0]?.id}/dashboard`)
+		}
+	}, [organizationByIdData])
+
```

Comment:
I would completly remove the part to redirect from the useCallback function and move it to the useEffect.
Therefore you need to add the dependency: "orgId" and check if it's set.

Smth. like this:
`useEffect(() => { if (orgId && organizationByIdData && !loading) {  ...redirect } }, [ orgId, organizationById])`

So we don't have the code twice.

Reply:
yes noted